src/formatting/comment.rs

```diff
@@ -1604,7 +1604,15 @@ pub(crate) fn recover_comment_removed(
     let snippet = context.snippet(span);
     let includes_comment = contains_comment(snippet);
     if snippet != new && includes_comment && changed_comment_content(snippet, &new) {
-        Some(snippet.to_owned())
+        /* Trim white spaces at end of lines */
+        let mut c = String::from("");
+        for line in snippet.to_owned().split('\n') {
+            if c != "" {
+                c.push('\n')
+            };
+            c.push_str(line.trim_end());
+        }
+        Some(c.to_owned())
```

Comment:
You've got these same changes in three separate PRs. Please keep various commits and changes separate for easier review and potential for merging. 

As I noted in https://github.com/rust-lang/rustfmt/pull/4391#issuecomment-679329947, I do not believe this is the right approach to address the related issue.